### PR TITLE
Adding logs documentation

### PR DESCRIPTION
**Pull request by alavieille**

### CODE REVIEW

File: en/developer_guide/logs.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+Logs
+====
+
+For logs we use Monolog. The Logs table is displayed in administration section in the backoffice. it displays the date and time, user's IP, user's name, the current site's name and the log's message.
+
+The LogBundle is in openorchestra-cmsbundle.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: This sentence is partially wrong.

By saying it, you put a limit on the refactoring we could do with the logBundle

File: en/developer_guide/logs.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
+Logs
+====
+
+For logs we use Monolog. The Logs table is displayed in administration section in the backoffice. it displays the date and time, user's IP, user's name, the current site's name and the log's message.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: Our backoffice logging is based on Monolog. We provide an interface to display all the user action on the website

File: en/developer_guide/logs.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Where the variable (node_id, node_version, node_language) are defined in the context, they are replaced when displaying.
+
+The different events that write the logs are in ``LogBundle\EventSubscriber``.
+
+Display
+-------
+
+The view who display logs is ``BackofficeBundle\Resources\views\AdministrationPanel\logs.html.twig``.
+
+The facades and transformers in ``LogBundle\Facade`` and ``LogBundle\Transformer``.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: are located in

File: en/developer_guide/logs.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    Update a node with node id node_id, node version node_version and node language node_language
+
+Where the variable (node_id, node_version, node_language) are defined in the context, they are replaced when displaying.
+
+The different events that write the logs are in ``LogBundle\EventSubscriber``.
+
+Display
+-------
+
+The view who display logs is ``BackofficeBundle\Resources\views\AdministrationPanel\logs.html.twig``.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: which displays

File: en/developer_guide/logs.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+====
+
+For logs we use Monolog. The Logs table is displayed in administration section in the backoffice. it displays the date and time, user's IP, user's name, the current site's name and the log's message.
+
+The LogBundle is in openorchestra-cmsbundle.
+
+Configuration
+-------------
+
+The logs's configuration is in app/config/log.yml. This file allows to declare the handlers. The handlers configure where save the logs the information type to record and channels.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: "where to save"
I think there is a comma missing in the last sentence.
This line is too long and should be split.

File: en/developer_guide/logs.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+To save the logs in MongoDB, We need to define a handler with connection's information of the base and collection's name. Level indicates information's type to record in the logs. For example 200 it's for info but there is also debug, error, etc ...
```
Comment (juchi) [inline en/developer_guide/logs.rst]: This line is too long and should be split

File: en/developer_guide/logs.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    tags:
+        - { name: monolog.processor, method: processRecord, channel: openorchestra }
+
+Logger
+------
+Services that write the logs must have in parameter ``Symfony\Bridge\Monolog\Logger``
+and tagged by ``monolog.logger``
+
+For example :
```
Comment (juchi) [inline en/developer_guide/logs.rst]: for instance

File: en/developer_guide/logs.rst [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+The processor must be tagged by:
+
+.. code-block:: yaml
+
+    tags:
+        - { name: monolog.processor, method: processRecord, channel: openorchestra }
+
+Logger
+------
+Services that write the logs must have in parameter ``Symfony\Bridge\Monolog\Logger``
```
Comment (juchi) [inline en/developer_guide/logs.rst]: services that write the logs take a `Symfony\Bridge\Monolog\Logger` in parameter and must be tagged by `monolog.logger``
It should be explicitly said that these services are event subscriber

File: en/developer_guide/logs.rst [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+.. code-block:: yaml
+
+    open_orchestra_log.subscriber.node:
+        class: %open_orchestra_log.subscriber.node.class%
+        arguments: [@logger]
+        tags:
+            - { name: kernel.event_subscriber }
+            - { name: monolog.logger, channel: openorchestra }
+
+On OpenOrchestra when an action like updating a node is done an event is created. It's this event who writes the logs messages.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Open Orchestra
The event is not the one writing the logs, it triggers it so a subscriber will log
who -> which
logs messages -> log messages

File: en/developer_guide/logs.rst [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+A log message takes two parameters, a translation key and a context. The messages are in the translation file for example :
+
+.. code-block:: yaml
+
+    Update a node with node id node_id, node version node_version and node language node_language
```
Comment (juchi) [inline en/developer_guide/logs.rst]: keys are missing for this to be yaml

File: en/developer_guide/logs.rst [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+Where the variable (node_id, node_version, node_language) are defined in the context, they are replaced when displaying.
+
+The different events that write the logs are in ``LogBundle\EventSubscriber``.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Again, events do not write logs

File: en/developer_guide/logs.rst [hunk 12]
```diff
@@ -0,0 +1,89 @@
+Logs
+====
+
+For logs we use Monolog. The Logs table is displayed in administration section in the backoffice.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: This sentence is partially wrong.

By saying it, you put a limit on the refactoring we could do with the logBundle

File: en/developer_guide/logs.rst [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+It displays the date and time, user's IP, user's name, the current site's name and the log's message.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: Our backoffice logging is based on Monolog. We provide an interface to display all the user action on the website

File: en/developer_guide/logs.rst [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+The LogBundle is in openorchestra-cmsbundle.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: This sentence is partially wrong.

By saying it, you put a limit on the refactoring we could do with the logBundle

File: en/developer_guide/logs.rst [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+Configuration
+-------------
+
+The logs's configuration is in app/config/log.yml. This file allows to declare the handlers.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: is located in the `app/config/log.yml` file

File: en/developer_guide/logs.rst [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+The handlers configure where save the logs the information type to record and channels.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: Your description is a little too procedural I think, you should rather start by the end : 

Monolog uses channels to record the log. To link a channel and a record, you will have to define an handler. The configuration of the handler can be placed in the `app/config/log.yml` file.

File: en/developer_guide/logs.rst [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+            type: mongo
+            level: 200
+            mongo:
+                host: "localhost"
+                database: "%open_orchestra_cms.mongodb.database%"
+                collection: log
+            channels: [openorchestra]
+
+
+Channels defined the channels where are written the logs.
```
Comment (ngilain) [inline en/developer_guide/logs.rst]: define

File: en/developer_guide/logs.rst [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+If no channel is indicated then the logs are written in all channels.
+It's possible to put multiple channels or exclude one.
```
Comment (ngilain) [inline en/developer_guide/logs.rst:35]: Can you put an example on how to exclude a channel ?

File: en/developer_guide/logs.rst [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+The services who have the channel ``openorchestra`` write only in this chanel
+and our handler shows logs who are in the ``openorchestra`` channel.
+
+Processor
+---------
+
+The loggers who that allow record logs have the date and time but not IP address and user name.
```
Comment (ngilain) [inline en/developer_guide/logs.rst]: The loggers allowing record

File: en/developer_guide/logs.rst [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+For save IP address and user name, we have created a processor.
```
Comment (ngilain) [inline en/developer_guide/logs.rst]: To save IP address [...]

Comment (ngilain) [inline en/developer_guide/logs.rst]: Avoid speaking of the team using "we", instead talk about what the product has : "Open Orchestra has a processor"

File: en/developer_guide/logs.rst [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+In the processor tags, we spent the channel ``openorchestra``.
+So, for each log written in this channel he spends into the processor.
```
Comment (ngilain) [inline en/developer_guide/logs.rst]: This sentence is not very clear

File: en/developer_guide/logs.rst [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+A log message takes two parameters, a translation key and a context.
+The messages are located in the translation file for example :
+
+.. code-block:: yaml
+
+    open_orchestra_log:
+        node:
+            update: Update a node with node id node_id, node version node_version and node language node_language
+
+Where the variable (node_id, node_version, node_language) are defined in the context,
```
Comment (ngilain) [inline en/developer_guide/logs.rst]: Could you rephrase this ? I think there's something wrong

**Comment by juchi** [comment]
As pointed out by Noel, there are several active voice sentences that should be rephrased to be passive.

File: en/developer_guide/logs.rst [hunk 23]
```diff
@@ -0,0 +1,90 @@
...
+====
+
+Our backoffice logging is based on Monolog.
+We provide an interface to display all the user action on the website
+It displays the date and time, user's IP, user's name, the current site's name and the log's message.
+
+Configuration
+-------------
+
+Monolog uses channels to record the log. To link a channel and a record,you will have to define an handler.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Space after comma

File: en/developer_guide/logs.rst [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+The configuration of the handler can be placed in the app/config/log.yml file.
+
+To save the logs in MongoDB, We need to define a handler
+with connection's information of the base and collection's name.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: with connection information to the database and the MongoDB collection's name.

File: en/developer_guide/logs.rst [hunk 25]
```diff
@@ -0,0 +1,90 @@
+Logs
+====
+
+Our backoffice logging is based on Monolog.
+We provide an interface to display all the user action on the website
+It displays the date and time, user's IP, user's name, the current site's name and the log's message.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: I think this sentence is too much, the one before can only invite the user to take a look at it.

Maybe you can add a screen shot of some lines

File: en/developer_guide/logs.rst (hunk 23 above)
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: you are missing a white space after the comma

File: en/developer_guide/logs.rst [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+
+Configuration
+-------------
+
+Monolog uses channels to record the log. To link a channel and a record,you will have to define an handler.
+The configuration of the handler can be placed in the app/config/log.yml file.
+
+To save the logs in MongoDB, We need to define a handler
+with connection's information of the base and collection's name.
+Level indicates information's type to record in the logs.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: The level entry indicates the information type to record.

File: en/developer_guide/logs.rst [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+
+Our backoffice logging is based on Monolog.
+We provide an interface to display all the user action on the website
+It displays the date and time, user's IP, user's name, the current site's name and the log's message.
+
+Configuration
+-------------
+
+Monolog uses channels to record the log. To link a channel and a record,you will have to define an handler.
+The configuration of the handler can be placed in the app/config/log.yml file.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: the double quotes are missing around the file name

File: en/developer_guide/logs.rst [hunk 28]
```diff
@@ -0,0 +1,90 @@
...
+
+To save the logs in MongoDB, We need to define a handler
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: do not put a W in upper case

Comment (nicolasThal) [inline en/developer_guide/logs.rst]: This configuration is specific to our installation with mongo.

Can you rather say, if you want to store the logs in mongo, you will have to define a handler with the database connection information.

Comment (nicolasThal) [inline en/developer_guide/logs.rst]: Never use `We` in a documentation, I will recommand you to prefer the use of `you`

File: en/developer_guide/logs.rst [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+with connection's information of the base and collection's name.
+Level indicates information's type to record in the logs.
+For example 200 it's for info but there is also debug, error, etc, ...
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Too imprecise, we should explain how the level is chosen :
"Each level is a number which value is standardized in the PSR-3 recommendation."

File: en/developer_guide/logs.rst (hunk 18 above)
Comment (nicolasThal) [inline en/developer_guide/logs.rst:35]: Can you put a link to the monolog documentation for further information?

File: en/developer_guide/logs.rst (hunk 17 above)
Comment (juchi) [inline en/developer_guide/logs.rst]: "The `channels key`"

**Comment by juchi** [comment]
I think a link to the monolog documentation would be appropriate, as many of the monolog concepts are not explained here.

File: en/developer_guide/logs.rst (hunk 19 above)
Comment (juchi) [inline en/developer_guide/logs.rst]: "The loggers allowing to record"
"have the date and time" -> "know the date and time" ?
the IP address

File: en/developer_guide/logs.rst [hunk 30]
```diff
@@ -0,0 +1,90 @@
...
+It's possible to put multiple channels or exclude one.
+The services who have the channel ``openorchestra`` write only in this chanel
+and our handler shows logs who are in the ``openorchestra`` channel.
+
+Processor
+---------
+
+The loggers who that allow record logs have the date and time but not IP address and user name.
+For save IP address and user name, we have created a processor.
+In the processor tags, we spent the channel ``openorchestra``.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: what do you mean by " spent" ? did you mean "sent" ?

File: en/developer_guide/logs.rst [hunk 31]
```diff
@@ -0,0 +1,90 @@
...
+So, for each log written in this channel he spends into the processor.
+
+The processor must be tagged by:
```
Comment (juchi) [inline en/developer_guide/logs.rst]: why _must_ it be tagged ? it should be explained.
Also, does the "method" need to match something particular ?

File: en/developer_guide/logs.rst [hunk 32]
```diff
@@ -0,0 +1,90 @@
...
+
+.. code-block:: yaml
+
+    tags:
+        - { name: monolog.processor, method: processRecord, channel: openorchestra }
+
+Logger
+------
+
+Services that write the logs are event subscriber. They must have in parameter ``Symfony\Bridge\Monolog\Logger``
```
Comment (juchi) [inline en/developer_guide/logs.rst]: subscribers

File: en/developer_guide/logs.rst [hunk 33]
```diff
@@ -0,0 +1,90 @@
...
+.. code-block:: yaml
+
+    open_orchestra_log.subscriber.node:
+        class: %open_orchestra_log.subscriber.node.class%
+        arguments: [@logger]
+        tags:
+            - { name: kernel.event_subscriber }
+            - { name: monolog.logger, channel: openorchestra }
+
+On Open Orchestra when an action like updating a node is done an event is created.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: "In Open Orchestra an event is dispatched on most user actions (ie. creation, update of elements...)."

File: en/developer_guide/logs.rst [hunk 34]
```diff
@@ -0,0 +1,90 @@
...
+It's this event which triggers writing the log messages.
+A log message takes two parameters, a translation key and a context.
+The messages are located in the translation file for example :
```
Comment (juchi) [inline en/developer_guide/logs.rst]: i would just put the column after "file" : "in the translation file:"

File: en/developer_guide/logs.rst [hunk 35]
```diff
@@ -0,0 +1,90 @@
...
+        node:
+            update: Update a node with node id node_id, node version node_version and node language node_language
+
+Where the variable (node_id, node_version, node_language) are defined in the context,
+they are replaced when displaying.
+
+The different events are in ``LogBundle\EventSubscriber``.
+
+Display
+-------
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Open question :
Is this paragraph really usefull ? If we don't provide an explanation about how to change the template or add a Facade / Transformer, i'm not sure it's important to say it.
ping @nicolasThal @itkg-ngilain @itkg-nanne @itkg-bfouche

**Comment by alavieille** [comment]
@nicolasThal @itkg-ngilain  @juchi

File: en/developer_guide/logs.rst [hunk 36]
```diff
@@ -0,0 +1,97 @@
...
+
+.. image:: ../../images/log_preview.png
+
+Configuration
+-------------
+
+Monolog uses channels to record the log. To link a channel and a record, you will have to define an handler.
+The configuration of the handler can be placed in the ``app/config/log.yml`` file.
+
+For instance, if you want to store the logs in mongo, you will have to define a handler with the database connection information.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: I would say `database connection parameters`

File: en/developer_guide/logs.rst [hunk 37]
```diff
@@ -0,0 +1,97 @@
...
+The level entry indicates the information type to record in the logs.
+Each level is a number which value is standardized in the PSR-3 recommendation.
+For example 200 it's for info. The different level are defined in the `Monolog documentaion`_
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: For instance

File: en/developer_guide/logs.rst [hunk 38]
```diff
@@ -0,0 +1,97 @@
...
+            type: mongo
+            level: 200
+            mongo:
+                host: "localhost"
+                database: "%open_orchestra_cms.mongodb.database%"
+                collection: log
+            channels: [openorchestra]
+
+
+The channels key defines the channels where are written the logs.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: `where the log are written`.

Instead of written, I would say published, because the log are only published, then there are handler to listen to a channel and write them to the database or file

File: en/developer_guide/logs.rst [hunk 39]
```diff
@@ -0,0 +1,97 @@
...
+It's possible to put multiple channels or exclude one.
+The services who have the channel ``openorchestra`` write only in this chanel
+and our handler shows logs who are in the ``openorchestra`` channel.
+
+The different possibility of configuration about channels are explained in the `channels documentation`_.
+
+Processor
+---------
+
+The loggers allowing to record logs know the date and time but not the IP address and user name.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: I think that you should also put a link to the monolog processor documentation

Comment (nicolasThal) [inline en/developer_guide/logs.rst]: `The loggers allowing`, i think that there is a grammatical error

File: en/developer_guide/logs.rst [hunk 40]
```diff
@@ -0,0 +1,97 @@
...
+To save IP address and user name, Open Orchestra has a processor.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: I would more say : 

`When you publish your log, some information may be missing. Monolog allows you to add a processor which will fill up the logging information by adding some context.
To do it, you can create a service, tagged by monolog.processor. You can referer to the Monolog processor documentation for further exemple.`

File: en/developer_guide/logs.rst [hunk 41]
```diff
@@ -0,0 +1,97 @@
...
+A processor is a service tagged by monolog.processor that allow adding extra data in the records.
+In the processor tags, Open Orchestra use the channel ``openorchestra``.
+
+So, for each log written in this channel, The method ``processRecord`` of processor is called , she adds the IP address in the recording.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: this sentence has some gramatical issue.

I would suggest you to delete it

File: en/developer_guide/logs.rst [hunk 42]
```diff
@@ -0,0 +1,97 @@
...
+
+.. code-block:: yaml
+
+    tags:
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: this code block is not usefull because we are just monolog users. A link to the monolog documentation will be more powerfull

File: en/developer_guide/logs.rst (hunk 37 above)
Comment (juchi) [inline en/developer_guide/logs.rst]: The sentence actually doesn't feel right.
"For instance 200 is the value for the "info" level." would sound better.
"The different levels"

File: en/developer_guide/logs.rst (hunk 38 above)
Comment (juchi) [inline en/developer_guide/logs.rst]: I would display "channels" with the code format as it refers to yaml content.

File: en/developer_guide/logs.rst [hunk 43]
```diff
@@ -0,0 +1,97 @@
...
+            level: 200
+            mongo:
+                host: "localhost"
+                database: "%open_orchestra_cms.mongodb.database%"
+                collection: log
+            channels: [openorchestra]
+
+
+The channels key defines the channels where are written the logs.
+If no channel is indicated then the logs are written in all channels.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: "specified" instead of "indicated"

File: en/developer_guide/logs.rst [hunk 44]
```diff
@@ -0,0 +1,97 @@
...
+It's possible to put multiple channels or exclude one.
+The services who have the channel ``openorchestra`` write only in this chanel
```
Comment (juchi) [inline en/developer_guide/logs.rst]: which

File: en/developer_guide/logs.rst [hunk 45]
```diff
@@ -0,0 +1,97 @@
...
+and our handler shows logs who are in the ``openorchestra`` channel.
+
+The different possibility of configuration about channels are explained in the `channels documentation`_.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: possibilities

File: en/developer_guide/logs.rst [hunk 46]
```diff
@@ -0,0 +1,97 @@
...
+
+.. code-block:: yaml
+
+    tags:
+        - { name: monolog.processor, method: processRecord, channel: openorchestra }
+
+Logger
+------
+
+The Services that write the logs are event subscribers. They must have into their parameter ``Symfony\Bridge\Monolog\Logger``
```
Comment (juchi) [inline en/developer_guide/logs.rst]: services takes a small "s"
"They must have into their parameter" is not correct

File: en/developer_guide/logs.rst [hunk 47]
```diff
@@ -0,0 +1,97 @@
...
+The messages are located in the translation file for example :
+
+.. code-block:: yaml
+
+    open_orchestra_log:
+        node:
+            update: Update a node with node id node_id, node version node_version and node language node_language
+
+In this example, the variables (node_id, node_version, node_language) are defined in the context,
+they are replaced when displaying.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Something like "on display" or "during log generation" would be better.

File: en/developer_guide/logs.rst [hunk 48]
```diff
@@ -0,0 +1,97 @@
...
+
+    open_orchestra_log.subscriber.node:
+        class: %open_orchestra_log.subscriber.node.class%
+        arguments: [@logger]
+        tags:
+            - { name: kernel.event_subscriber }
+            - { name: monolog.logger, channel: openorchestra }
+
+In Open Orchestra an event is dispatched on most user actions (ie. creation, update of elements...).
+It's this event which triggers writing the log messages.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: `This event publish the log message in the right channel`.

**Comment by alavieille** [comment]
@juchi @nicolasThal @itkg-ngilain

File: en/developer_guide/logs.rst [hunk 49]
```diff
@@ -0,0 +1,92 @@
...
+and our handler shows logs who are in the ``openorchestra`` channel.
+
+The different possibilities of configuration about channels are explained in the `channels documentation`_.
+
+Processor
+---------
+When you publish your log, some information may be missing.
+Monolog allows you to add a processor which will fill up the logging information by adding some context.
+
+To do it, you can create a service, tagged by monolog.processor.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: `tag it with` instead of tagged by

File: en/developer_guide/logs.rst [hunk 50]
```diff
@@ -0,0 +1,92 @@
...
+You can refer to the Monolog `processor documentation`_ for further example.
+
+Logger
+------
+
+The services that write the logs are event subscribers. They need ``Symfony\Bridge\Monolog\Logger`` in their parameter
+and be tagged by ``monolog.logger``
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: `and tag it with`

File: en/developer_guide/logs.rst (hunk 35 above)
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: I agree with @juchi .

This part will be more usefull in the api documentation

**Comment by alavieille** [comment]
@nicolasThal @juchi @itkg-ngilain

File: en/developer_guide/logs.rst (hunk 35 above)
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: @itkg-alavieille you can suppress the display ppart

File: en/developer_guide/logs.rst [hunk 51]
```diff
@@ -0,0 +1,92 @@
...
+                database: "%open_orchestra_cms.mongodb.database%"
+                collection: log
+            channels: [openorchestra]
+
+
+The ``channels`` key defines the channels where the log are published.
+If no channel is specified then the logs are written in all channels.
+It's possible to put multiple channels or exclude one.
+The services which have the channel ``openorchestra`` write only in this chanel
+and our handler shows logs who are in the ``openorchestra`` channel.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: which

File: en/developer_guide/logs.rst [hunk 52]
```diff
@@ -0,0 +1,92 @@
...
+
+The different possibilities of configuration about channels are explained in the `channels documentation`_.
+
+Processor
+---------
+When you publish your log, some information may be missing.
+Monolog allows you to add a processor which will fill up the logging information by adding some context.
+
+To do it, you can create a service, tag it with monolog.processor.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: the tag name should be emphasised
"tagged with [...]"

File: en/developer_guide/logs.rst [hunk 53]
```diff
@@ -0,0 +1,92 @@
...
+
+    open_orchestra_log.subscriber.node:
+        class: %open_orchestra_log.subscriber.node.class%
+        arguments: [@logger]
+        tags:
+            - { name: kernel.event_subscriber }
+            - { name: monolog.logger, channel: openorchestra }
+
+In Open Orchestra an event is dispatched on most user actions (ie. creation, update of elements...).
+This event publish the log message in the right channel.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: publishes

File: en/developer_guide/logs.rst [hunk 54]
```diff
@@ -0,0 +1,92 @@
...
+A log message takes two parameters, a translation key and a context (data array about the message).
+The messages are located in the translation file for example :
```
Comment (juchi) [inline en/developer_guide/logs.rst]: comma missing, no space before column : "in the translation file, for instance:"

File: en/developer_guide/logs.rst [hunk 55]
```diff
@@ -0,0 +1,92 @@
...
+Our backoffice logging is based on `Monolog`_.
+Open Orchestra provides an interface to display all the user actions on the website.
+
+.. image:: ../../images/log_preview.png
+
+Configuration
+-------------
+
+Monolog uses channels to record the log. To link a channel and a record, you will have to define an handler.
+The configuration of the handler can be placed in the ``app/config/log.yml`` file.
```
Comment (juchi) [inline en/developer_guide/logs.rst]: I think this is misleading. The configuration can be placed anywhere, as long as it is included somewhere. If we eventually decide to not provide the full filesystem of the symfony app this will not be true.
@nicolasThal what do you think ?

File: en/developer_guide/logs.rst [hunk 56]
```diff
@@ -0,0 +1,92 @@
...
+When you publish your log, some information may be missing.
+Monolog allows you to add a processor which will fill up the logging information by adding some context.
+
+To do it, you can create a service, tag it with monolog.processor.
+You can refer to the Monolog `processor documentation`_ for further example.
+
+Logger
+------
+
+The services that write the logs are event subscribers. They need ``Symfony\Bridge\Monolog\Logger`` in their parameter
```
Comment (juchi) [inline en/developer_guide/logs.rst]: Still sounds grammatically incorrect.
I believe it should be rephrased to something like "They should be injected the [MonologLogger] in order to be functionnal and be tagged with monolog.logger."

**Comment by juchi** [comment]
Display section should be removed as agreed.

File: en/developer_guide/logs.rst (hunk 55 above)
Comment (alavieille) [inline en/developer_guide/logs.rst]: I can change the sentence by => Monolog uses channels to record the log. To link a channel and a record, you will have to define an handler in your configuration file.

Comment (nicolasThal) [inline en/developer_guide/logs.rst]: Yes, you can just say that you place the configuration in a configuration file

**Comment by alavieille** [comment]
@juchi @nicolasThal

File: en/developer_guide/logs.rst [hunk 57]
```diff
@@ -0,0 +1,85 @@
...
+
+    open_orchestra_log.subscriber.node:
+        class: %open_orchestra_log.subscriber.node.class%
+        arguments: [@logger]
+        tags:
+            - { name: kernel.event_subscriber }
+            - { name: monolog.logger, channel: openorchestra }
+
+In Open Orchestra an event is dispatched on most user actions (ie. creation, update of elements...).
+This event publishes the log message in the right channel.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: This event is catched to publish the log message in the right channel

File: en/developer_guide/logs.rst [hunk 58]
```diff
@@ -0,0 +1,85 @@
...
+Monolog allows you to add a processor which will fill up the logging information by adding some context.
+
+To do it, you can create a service, tag it with [monolog.processor].
+You can refer to the Monolog `processor documentation`_ for further example.
+
+Logger
+------
+
+The services that write the logs are event subscribers.
+They should be injected the [MonologLogger] in order to be functional and be tagged with [monolog.logger].
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: use `` instead of [ for the code part

File: en/developer_guide/logs.rst [hunk 59]
```diff
@@ -0,0 +1,85 @@
...
+The messages are located in the translation file, for instance:
+
+.. code-block:: yaml
+
+    open_orchestra_log:
+        node:
+            update: Update a node with node id node_id, node version node_version and node language node_language
+
+In this example, the variables (node_id, node_version, node_language) are defined in the context,
+they are replaced during log generation.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: that not totaly right, the replacement is made when you display the log on the backoffice. We only save the translation key and the context

File: en/developer_guide/logs.rst [hunk 60]
```diff
@@ -0,0 +1,85 @@
...
+
+The different events are in ``LogBundle\EventSubscriber``.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: The subscriber are in this bundle, not the events :)

**Comment by alavieille** [comment]
@nicolasThal

File: en/developer_guide/logs.rst [hunk 61]
```diff
@@ -0,0 +1,85 @@
...
+Monolog allows you to add a processor which will fill up the logging information by adding some context.
+
+To do it, you can create a service, tag it with ``monolog.processor``.
+You can refer to the Monolog `processor documentation`_ for further example.
+
+Logger
+------
+
+The services that write the logs are event subscribers.
+They should be injected the ``MonologLogger`` in order to be functional and be tagged with ``monolog.logger``.
```
Comment (nicolasThal) [inline en/developer_guide/logs.rst]: Can you put the real namespace of the class instead of MonologLogger, which does not refer to anything

**Comment by alavieille** [comment]
@nicolasThal @juchi @itkg-ngilain

**Comment by nicolasThal** [comment]
:+1:

File: en/developer_guide/logs.rst [hunk 62]
```diff
@@ -0,0 +1,85 @@
...
+.. code-block:: yaml
+
+    open_orchestra_log:
+        node:
+            update: Update a node with node id node_id, node version node_version and node language node_language
+
+In this example, the variables (node_id, node_version, node_language) are defined in the context,
+they are replaced on display.
+
+The different subscribers are in ``LogBundle\EventSubscriber``.
```
Comment (itkg-bfouche) [inline en/developer_guide/logs.rst]: OpenOrchestra\LogBundle\EventSubscriber would be better

**Comment by itkg-bfouche** [comment]
:+1:  with OpenOrchestra\LogBundle\EventSubscriber

**Comment by nicolasThal** [comment]
Do not forget to squash the commits before merging!

**Comment by juchi** [comment]
:+1:
